### PR TITLE
feat: improve client-side field validation on Partners listings

### PR DESCRIPTION
**Pull request by jaredcwhite**
This PR addresses metrotranscom/doorway#771

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This PR sets up the Date/Time fields from UIC so that they can forbid non-numerical input. It also added a client-side validation trigger to the Listings field, which I tested a lot so it *shouldn't* add any weird edge cases to how saving a Listing works, but that should definitely have a bit of QA attention.

## How Can This Be Tested/Reviewed?

Add or edit a listing, and try to make sure you can't type letters and also nonsense numbers (like the year 400 or 23 for a month) for Date & Time fields like Application Due Date, Open House Dates, submitted time on Applications, etc.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

File: sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx [hunk 1]
```diff
@@ -211,6 +213,7 @@ const RankingsAndResults = ({ listing, isAdmin }: RankingsAndResultsProps) => {
                   name={"lotteryEndTime"}
                   id={"lotteryEndTime"}
                   register={register}
+                  setValue={setValue}
```
Comment (mcgarrye) [inline sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:216]: Previously if these fields were blank on publish/save, they would all be highlighted red with a warning. Now if they are left blank there is no red warning but an error still appears at the top. Not sure why only the date field is still correctly showing that error

**Review by mcgarrye — COMMENTED** [review]
Overall things are looking good, if you could just fix the above comment and check out why that unit test is failing this should be good to go

**Comment by jaredcwhite** [comment]
@mcgarrye Thanks for the review! I'll look into the issue you mentioned.

File: sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx (hunk 1 above)
Comment (jaredcwhite) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:219]: @mcgarrye Can you take one more look at this? I'm not entirely sure why it previously worked in main, but as those date & time fields are required, I wouldn't expect them to validate entry on save without there being a `required` prop. So I've added those and it seems to be acting OK on the PR to me now.

Comment (mcgarrye) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:219]: Screenshot 2024-10-18 at 12 23 56 PM
I'm still not seeing the red highlight on the fields :/

Comment (jaredcwhite) [reply sites/partners/src/components/listings/PaperListingForm/sections/RankingsAndResults.tsx:219]: @mcgarrye Ah, so it _is_ working on the date field, but not on the time fields if one of those is blank. Hmm, I will investigate!

**Comment by jaredcwhite** [comment]
@mcgarrye Third time's a charm?

**Review by mcgarrye — APPROVED** [review]
Looks good! Once you get the test passing lgtm

**Pull request merged**